### PR TITLE
Production test for fmcomms5

### PR DESCRIPTION
**Pull request by ranechita**
# Description

Production tests for fmcomms5 were added, similar to the ones existing in iio-oscilloscope. EEPROM-related functions and hardware gain test were updated to work with fmcomms5 as well.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Test Configuration**:
* Hardware: ADRV9361-ZC702
* OS: Ubuntu 20.04

# Documentation

If this is a new feature or example please mention or link any documentation. All new hardware interface classes require documentation.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: test/test_fmcomms2-3_p.py [hunk 1]
```diff
@@ -211,6 +211,6 @@ def test_dcxo(test_dcxo_calibration, classname, iio_uri):
     [([-20.0, -100.0, -110.0, -110.0, -110.0], [-10.0, -60.0, -75.0, -75.0, -80.0])],
 )
 def test_harmonic_values(
-    test_harmonics, classname, iio_uri, channel, param_set, low, high, plot=False
+    test_harmonics, classname, iio_uri, channel, param_set, low, high, plot=True
```
Comment (tfcollins) [inline test/test_fmcomms2-3_p.py:214]: Plot should be False

File: test/test_fmcomms5_prod.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+            sample_rate=30720000,
+        )
+    ],
+)
+@pytest.mark.parametrize(
+    "low, high",
+    [([-20.0, -100.0, -110.0, -110.0, -110.0], [-10.0, -60.0, -75.0, -75.0, -80.0])],
+)
+def test_harmonic_values_chip_b(
+    test_harmonics, classname, iio_uri, channel, param_set, low, high, plot=True
```
Comment (tfcollins) [inline test/test_fmcomms5_prod.py:283]: Do you want the plots to show up here?

Comment (ranechita) [reply test/test_fmcomms5_prod.py:283]: In the case of production tests I think it is relevant to be able to see the plots by default, without having to edit the code

Comment (ranechita) [reply test/test_fmcomms5_prod.py:283]: I will make an update soon and remove them

**Pull request closed without merging**